pkg/storage/unified/sql/server.go

```diff
@@ -54,42 +54,24 @@ func NewResourceServer(ctx context.Context, db infraDB.DB, cfg *setting.Cfg,
 	opts.Backend = store
 	opts.Diagnostics = store
 	opts.Lifecycle = store
-	opts.Search = resource.SearchOptions{
-		Resources: docs,
-	}
 
+	// Setup the search server
 	if features.IsEnabledGlobally(featuremgmt.FlagUnifiedStorageSearch) {
-		opts.Index = resource.NewResourceIndexServer(cfg, tracer)
+		opts.Search = resource.SearchOptions{
+			Backend: search.NewBleveBackend(search.BleveOptions{
+				Root:          filepath.Join(cfg.DataPath, "unified-search", "bleve"),
+				FileThreshold: 10,  // fewer than X items will use a memory index
+				BatchSize:     500, // This is the batch size for how many objects to add to the index at once
+			}, tracer, reg),
+			Resources:     docs,
+			WorkerThreads: 5, // from cfg?
+		}
 	}
 
 	rs, err := resource.NewResourceServer(opts)
 	if err != nil {
 		return nil, err
 	}
 
-	// Initialize the indexer if one is configured
-	if opts.Index != nil {
-		// TODO: Create a proper identity for the indexer
-		orgId := int64(1)
-		ctx = identity.WithRequester(ctx, &identity.StaticRequester{
```

Comment:
^^^ indexer does not need a special identity anymore because it only receives values and never asks for them